core/src/main/scala/kafka/cluster/Partition.scala

```diff
@@ -499,7 +500,16 @@ class Partition(val topicPartition: TopicPartition,
         addingReplicas = addingReplicas,
         removingReplicas = removingReplicas
       )
-      createLogIfNotExists(partitionState.isNew, isFutureReplica = false, highWatermarkCheckpoints)
+      try {
+        this.createLogIfNotExists(partitionState.isNew, isFutureReplica = false, highWatermarkCheckpoints)
+      } catch {
+        case e: ZooKeeperClientException =>
+          stateChangeLogger.info(s"A ZooKeeper client exception has occurred and makeLeader will be skipping the " +
+            s"state change for the partition with leader epoch: $leaderEpoch ", e)
+          error(s"ZooKeeper client error occurred while this partition was becoming the leader for $topicPartition.", e)
```

Comment:
We probably can just keep the state change logging. Also, the logging level probably should be error instead of info. Ditto below.

Reply:
Thanks @junrao -- I will make the logging level to error. 
Could you clarify what you mean by keep the state change logging?

Reply:
I meant removing the line in 509 of error(s"ZooKeeper client .

Reply:
Makes sense. Would appreciate a review on the change -- @junrao .